### PR TITLE
perform a dry-run create of Certificate for cert-manager check 

### PR DESCRIPTION
**Pull request by xinredhat**
**Changes:**
Added one check that attempts to perform a dry-run create of a cert-manager Certificate resource in order to verify that CRDs are installed and all the required webhooks are reachable by the K8S API server.

### CODE REVIEW

File: ckcp/openshift_dev_setup.sh [hunk 1]
```diff
@@ -178,12 +178,33 @@ install_cert_manager() {
...
+  # Wait until cert-manager pods are ready
   pods_cmd="kubectl -n openshift-cert-manager get pods -o=jsonpath='{range .items[*]}{.metadata.name}{\" \"}{.status.containerStatuses[*].ready}{\"\n\"}{end}'"
   until [[ $(eval $pods_cmd | wc -l) -eq 3 ]]; do
     echo -n "."
     sleep 5
   done
+
+  # perform a dry-run create of a cert-manager
+  # Certificate resource in order to verify that CRDs are installed and all the
+  # equired webhooks are reachable by the K8S API server.
```
Comment (Roming22) [inline ckcp/openshift_dev_setup.sh:190]: `equired` -> `requried`

File: ckcp/openshift_dev_setup.sh [hunk 2]
```diff
@@ -178,12 +178,33 @@ install_cert_manager() {
...
+  local tmp_dir=$(mktemp -d)
+  echo "
```
Comment (Roming22) [inline ckcp/openshift_dev_setup.sh:192]: I'm not a big fan of in-lining resources in scripts. Maybe you could create the yaml in `ckcp/openshift-operators/cert-manager/`? That would get rid of the tmp_dir as well.

File: ckcp/openshift_dev_setup.sh (hunk 1 above)
Comment (xinredhat) [reply ckcp/openshift_dev_setup.sh]: fixed

File: ckcp/openshift_dev_setup.sh (hunk 2 above)
Comment (xinredhat) [reply ckcp/openshift_dev_setup.sh:192]: The current in-lining resource of  is more like a temporary  resource, it's useless for any applications. It doens't make sense to create the yaml in  ckcp/openshift-operators/cert-manager/

Comment (Roming22) [reply ckcp/openshift_dev_setup.sh:192]: Then the tmp_dir should be cleaned up .

Comment (xinredhat) [reply ckcp/openshift_dev_setup.sh:192]: @Roming22 I would like to use this file `ckcp/openshift/base/certs.yaml` for executing dry-run check. WDYT? So  we don't need the in-lining resources in scripts

Comment (xinredhat) [reply ckcp/openshift_dev_setup.sh]: fixed.

**Comment by bnallapeta** [comment]
@xinredhat @Roming22 with k8s 1.24, --dry-run must be used with client or server. It throws this error otherwise:

```
[bnr@bnr pipelines-service]$ kubectl create -f /home/bnr/workspace/pipelines-service/ckcp/openshift/base/certs.yaml --dry-run
error: --dry-run flag without a value was specified. A value must be set: "none", "server", or "client".
```

have raised #161 to fix this.

**Pull request merged**